src/filters/tls.rs

```diff
@@ -0,0 +1,24 @@
+//! Tls filters
+
+use std::convert::Infallible;
+
+use crate::filter::{filter_fn_one, Filter};
+
+/// Creates a `Filter` to get the remote tls certificate chain of the connection.
+///
+/// If the underlying connection doesn't have certificates, this will yield
+/// `None`.
+///
+/// # Example
+///
+/// ```
+/// use warp::Filter;
+///
+/// let route = warp::::tls::peer_certificates()
+///     .map(|certs: Option<Vec<Vec<u8>>>| {
+///         println!("peer certificates = {:?}", certs);
+///     });
+/// ```
+pub fn peer_certificates() -> impl Filter<Extract = (Option<Vec<Vec<u8>>>,), Error = Infallible> + Copy {
```

Comment:
I wonder if this should be a stronger type. Maybe `Vec<Certificate>`, with `struct Certificate(Vec<u8>)`?

Reply:
yeah I recall now you mentioned it on the issue, Should it also have methods to return it's inner, like `get_ref` or `into_inner`?